xonsh/aliases.py

```diff
@@ -75,7 +76,7 @@ def source_foreign(args, stdin=None):
     ns = parser.parse_args(args)
     if ns.prevcmd is not None:
         pass  # don't change prevcmd if given explicitly
-    elif os.path.isfile(ns.files_or_code[0]):
+    elif os.path.isfile(ns.files_or_code[0]) and args[0] != 'cmd':
```

Comment:
Well. This line (actually the one below) wraps the file name of the script we source in quotes. That causes the subprocess to fail. I just can't figure out why. The same command work when I run it manually in cmd.exe.